pandas/core/internals/managers.py

```diff
@@ -1220,8 +1220,81 @@ def value_getitem(placement):
             if inplace and blk.should_store(value):
                 # Updating inplace -> check if we need to do Copy-on-Write
                 if using_copy_on_write() and not self._has_no_reference_block(blkno_l):
-                    blk.set_inplace(blk_locs, value_getitem(val_locs), copy=True)
-                    self._clear_reference_block(blkno_l)
+                    leftover_blocks = tuple(blk.delete(blk_locs))
+                    vals = np.broadcast_to(
```

Comment:
If new_block_2d doesn't copy vals, will need to make a copy since output of np.broadcast_to isn't writable.